src/iterative_ensemble_smoother/utils.py

```diff
@@ -80,7 +80,7 @@ def _create_errors(
     if len(observation_errors.shape) == 2:
         R = observation_errors
         observation_errors = np.sqrt(observation_errors.diagonal())
-        R = (R.T / R.diagonal()).T
+        R = np.diag(1 / observation_errors) @ R @ np.diag(1 / observation_errors)
```

Comment:
Not sure I understand this.
What will the dimension of `R` be after transformation if for example `observation_errors` is a 2x2 matrix?
In this case, I think `R` will end up just being a scalar, but I may very well be wrong.

Reply:
The dimensions should be the same as the input covariance matrix, I believe, given that all matrices involved are square with the same dimension.

Let $\Sigma\in R^{n\times n}$, and  $\Lambda = \text{diag}(\sigma)\in R^{n\times n}$ the diagonal matrix with standard deviations along its diagonal, then the correlation matrix is given by $\Lambda^{-1}\Sigma\Lambda^{-1}\in R^{n\times n}$.
`np.diag(1 / observation_errors)` is the inverse of the diagonal matrix with standard deviations.

The correlation matrix has elements $Cov(x_i, x_j)/(\sigma_i \sigma_j)$ which is what will be obtained when scaling the noise or error matrices, like $E$, with its standard deviations. This because we obtain a covariance matrix through $EE^T$, where the transpose makes sure $\sigma_i \sigma_j$ arrives in every denominator.

Why we seek to scale the different matrices as such is beyond me (but I think numerical stability?) since I have not yet read the inversion code yet, but this transform should correspond to how we scale `E` and `D` with the standard deviations in `observation_errors`.

Reply:
A little sketch creating a sample covariance matrix
<img width="739" alt="image" src="https://user-images.githubusercontent.com/13650516/228464209-d6983925-e305-4c94-86f3-22a4c97d409a.png">

Reply:
But `observation_errors` has been modified to a 1D array of the standard deviations, previous to the scaling of `R`
```
observation_errors = np.sqrt(observation_errors.diagonal())
```
So I still think everything should be okay.

Reply:
Copied over the code exactly here and print `R` and its dimensions
<img width="778" alt="image" src="https://user-images.githubusercontent.com/13650516/228496072-158c0534-0f2a-45c1-86f2-0f817d258bce.png">

Reply:
> But `observation_errors` has been modified to a 1D array of the standard deviations, previous to the scaling of `R`
> 
> ```
> observation_errors = np.sqrt(observation_errors.diagonal())
> ```
> 
> So I still think everything should be okay.

Ah, you are right, I did not read the that line 👍